modules/programs/gradle.nix

```
@@ -0,0 +1,134 @@
+        default = null;
+        description = ''
+          Path of the init script file to use.
+          If the text option is set, it will be preferred.
+        '';
+      };
+    };
+  });
+in {
+  meta.maintainers = [ maintainers.britter ];
+
+  options.programs.gradle = {
+    enable = mkEnableOption "Gradle Build Tool";
+    home = mkOption {
+      type = types.str;
+      default = defaultHomeDirectory;
+      description = ''
+        The Gradle home directory, relative to `home.homeDirectory`.
+
+        If set, the GRADLE_USER_HOME environment variable will be
+        set accordingly. Defaults to .gradle.
+      '';
+    };
+    package = mkPackageOption pkgs "gradle" { example = "pkgs.gradle_7"; };
+    settings = mkOption {
+      type = types.submodule { freeformType = settingsFormat.type; };
+      default = { };
+      example = {
+        "org.gradle.caching" = true;
+        "org.gradle.parallel" = true;
+        "org.gradle.jvmargs" = "-XX:MaxMetaspaceSize=384m";
+        "org.gradle.home" = pkgs.jdk17;
+      };
+      description = ''
+        Key value pairs to write to gradle.properties in the Gradle home directory.
+      '';
+    };
+    initScripts = mkOption {
+      type = types.either types.path (types.listOf initScript);
+      default = [ ];
+      example = literalExpression ''
+        [
+          {
+            name = "maven-local.gradle"
+            text = '''
+              allProject {
+                repositories {
+                  mavenLocal()
+                }
+              }
+            ''';
+          }
+          {
+            name = "another.init.gradle.kts"
+            source = ./another.init.gradle.kts
+          }
+        ]
+      '';
+      description = ''
+        Definition of init scripts to link into the Gradle home directory.
+
+        This can either be a path pointing to a directory containing init
+        scripts, or a list of init script definitions.
+        If a path is specified, that directory will be linked to the init.d
+        inside the Gradle Home directory.
+        If script definitions are specified they will be put into a directoy
+        that is then linked to init.d inside the Gradle Home directoy.
+
+        For more information about init scripts, including naming conventions
+        see https://docs.gradle.org/current/userguide/init_scripts.html.
+      '';
+    };
+  };
+
+  config = let
+    initScripts = if builtins.isPath cfg.initScripts then
+      cfg.initScripts
+    else
+      let
+        initScriptDirectory = pkgs.symlinkJoin {
+          name = "gradle-init-scripts";
+          paths =
+            map (script: pkgs.writeTextDir "init.d/${script.name}" script.text)
+            cfg.initScripts;
+        };
+      in "${initScriptDirectory}/init.d";
+    gradleHome = "${config.home.homeDirectory}/${cfg.home}";
+  in mkIf cfg.enable {
+    home.packages = [ cfg.package ];
+    home.file."${cfg.home}/init.d" =
```

Can add an empty line between the bindings to make it less compact:
```suggestion

    home.file."${cfg.home}/init.d" =
```